style_guide.rst

```diff
@@ -0,0 +1,393 @@
+
+- Use a space after these keywords::
+
+	if, switch, case, for, do, while
+
+  Yes:
+
+  ::
+
+       if (x == 5)
+
+  No:
+
+  ::
+
+       if(x == 5)
+
+- Use spaces around arithmetic operators:
+
+  Yes:
+
+  ::
+
+        i = i + 1
+        submitted += 1
+        x = x * 2 - 1
+        hypot2 = x * x + y * y
+        c = (a + b) * (a - b)
+
+  No:
+
+  ::
+
+        i=i+1
+        submitted +=1
+        x = x*2 - 1
+        hypot2 = x*x + y*y
+        c = (a+b) * (a-b)
+
+- Compound statements (multiple statements on the same line) are generally discouraged.
+
+
+  Yes:
+
+  ::
+
+        if (x == 0)
+            do_blah_thing();
+        do_one();
+        do_two();
+        do_three();
+
+  Rather not:
+
+  ::
+
+        if (x == 0) do_blah_thing();
+        do_one(); do_two(); do_three();
+
+
+Comments
+--------
+
+Comments that contradict the code are worse than no comments. Always
+make a priority of keeping the comments up-to-date when the code
+changes!  Comments should consist of either a single short phrase or
+one or more complete sentences. The first word of a comment should be
+capitalized, unless it is an identifier that begins with a lower case
+letter (never alter the case of identifiers!).
+
+If a comment is short, the period at the end can be omitted. Block
+comments generally consist of one or more paragraphs built out of
+complete sentences, and each sentence should end in a period.
+
+
+Header Comments
+^^^^^^^^^^^^^^^
+
+Header comments appear at the top of a file. These lines typically
+include the filename, author, date, version number, and a description
+of what the file is for and what it contains.
+
+::
+
+    /*
+     *  log.c: Logging functions
+     *
+     *  Use these functions to print log messages. Each message has an
+     *  associated log level:
+     *
+     *  CRITICAL: A critical unrecoverable error
+     *  ERROR: A recoverable error
+     *  WARNING: A warning
+     *  INFO: High-level information about the progress of the application
+     *  DEBUG: Lower-level information
+     *  TRACE: Very low-level information.
+     *
+     */
+
+
+Function Comments
+^^^^^^^^^^^^^^^^^
+
+Function comments should be done in the form of a multiline comment
+above the function header.
+
+This comment must contain information specific to what a function does.
+It should also include a description of the purpose and expected
+input arguments, the expected output values, and how error conditions
+are handled.
+
+Example:
+
+::
+
+    /*
+     * chilog - Print a log message
+     *
+     * level: Logging level of the message
+     *
+     * fmt: printf-style formatting string
+     *
+     * ...: Extra parameters if needed by fmt
+     *
+     * Returns: nothing.
+     */
+    void chilog(loglevel_t level, char *fmt, ...);
+
+
+Block Comments
+^^^^^^^^^^^^^^
+
+Block comments (``/* ... */``) generally apply to some (or all) code that follows
+them, and are indented to the same level as that code.
+
```

Comment:
Can we add suggestions for where these should go in if/else statements? 
i.e. should it be
<pre><code> if (condition) {
     /* comment */
     code 
} else {
      /*  comment */
</code></pre>
or
<pre><code>/* comment */
if (condition) {
      code
/* comment */
} else {
     code
}
</code></pre>

Reply:
Added some text on documenting if-else statements.